tests/rust/test_rust_backend.py

```
@@ -0,0 +1,674 @@
+    '''
+    backend = RustBackend()
+    source_file = temp_dir / "benchmark.rs"
+    source_file.write_text(rust_code)
+    if hasattr(backend, 'compile'):
+        start = time.time()
+        try:
+            _ = backend.compile(str(source_file))
+            elapsed = time.time() - start
+            assert elapsed < 10.0, f"Compilation took too long: {elapsed}s"
+        except Exception:
+            pytest.skip("Performance benchmark failed due to environment issues")
```

Ruff (0.11.9)

No newline at end of file

Add trailing newline

(W292)